src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ViewportManagerWithPlatformFeatures.cs

```diff
@@ -255,7 +255,9 @@ public override void OnLayoutChanged(bool isVirtualizing)
 			{
 				m_effectiveViewportChangedRevoker?.Dispose();
 			}
-			else if (m_effectiveViewportChangedRevoker == null)
+			else if (m_effectiveViewportChangedRevoker == null
+				// Uno workaround: [Perf] Do not listen for viewport update if nothing to render!
+				&& m_owner.ItemsSourceView?.Count > 0)
```

Comment:
Will this be called again if/when `ItemsSourceView.Count` becomes non-zero?

Reply:
Good question, and the answer is yes! The collection changed will trigger an `InvalidateMeasure` in all path, which will call the `OnBeginMeasure` on this class, which will call the `EnsureScroller()` (i.e. you second comment)